components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/internal/ClientManager.java

```diff
@@ -0,0 +1,202 @@
+                        .getHTTPConnectionTimeout())
+                .setConnectionRequestTimeout(WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                        .getHttpConnectionRequestTimeout())
+                .setSocketTimeout(
+                        WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration().getHttpReadTimeout())
+                .setRedirectsEnabled(false)
+                .setRelativeRedirectsAllowed(false)
+                .build();
+    }
+
+    private PoolingNHttpClientConnectionManager createPoolingConnectionManager() throws IOException {
+
+        int maxConnections = WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                .getDefaultMaxConnections();
+        int maxConnectionsPerRoute = WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                .getDefaultMaxConnectionsPerRoute();
+
+        ConnectingIOReactor ioReactor = new DefaultConnectingIOReactor();
+        PoolingNHttpClientConnectionManager poolingHttpClientConnectionMgr = new
+                PoolingNHttpClientConnectionManager(ioReactor);
+        // Increase max total connection to 20.
+        poolingHttpClientConnectionMgr.setMaxTotal(maxConnections);
+        // Increase default max connection per route to 20.
+        poolingHttpClientConnectionMgr.setDefaultMaxPerRoute(maxConnectionsPerRoute);
+        LOG.debug("PoolingNHttpClientConnectionManager created with maxConnections: " + maxConnections +
+                " and maxConnectionsPerRoute: " + maxConnectionsPerRoute);
+        return poolingHttpClientConnectionMgr;
+    }
+
+    private void addSslContext(HttpAsyncClientBuilder builder) throws WebSubAdapterException {
+
+        try {
+            SSLContext sslContext = SSLContexts.custom()
+            //default trust strategy is used (trusting all certificates in the provided trust store).
+                    .loadTrustMaterial(WebSubHubAdapterDataHolder.getInstance().getTrustStore(), null)
+                    .build();
+            builder.setSSLContext(sslContext);
+            builder.setSSLHostnameVerifier(new DefaultHostnameVerifier());
+            LOG.debug("SSL context and hostname verifier added");
+        } catch (NoSuchAlgorithmException | KeyStoreException | KeyManagementException e) {
+            throw WebSubHubAdapterUtil.handleServerException
+                    (WebSubHubAdapterConstants.ErrorMessages.ERROR_CREATING_SSL_CONTEXT, e);
+        }
+    }
+
+    /**
+     * Create an HTTP POST request.
+     *
+     * @param url The URL for the HTTP POST request.
+     * @param payload The payload to include in the request body.
+     * @return A configured HttpPost instance.
+     * @throws WebSubAdapterException If an error occurs while creating the request.
+     */
+    public HttpPost createHttpPost(String url, Object payload) throws WebSubAdapterException {
```

Comment:
Not formatted

Reply:
Addressed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/ec4d75aa968d6cbb73dd855225ccbadf7545d9fb